### ISSUE TITLE
Improve cilium trace output to be more clear 

### ISSUE THREAD

**Issue by techcet** [cilium/cilium#1610]
`cilium policy trace` output is somewhat confusing.
Improvement requested to remove redundancies ( `Result: ALLOWED` vs last line `Verdict: Allowed`). Redundancy caused by tracing output produced by agent via API, as well as the CLI tool itself (reference: @tgraf ).

Prepending the output of the final fate of the trace with "Final" could be useful, e.g. the last line could be "Final Verdict: xxx". 

```$ cilium policy trace -s id.curl -d id.httpd --dport 80
Tracing From: [container:id.curl] => To: [container:id.httpd] Ports: [80/any]
* Rule 2 {"matchLabels":{"any:id.httpd":""}}: match
    Allows from labels {"matchLabels":{"any:id.curl":""}}
+     Found all required labels
1 rules matched
Result: ALLOWED
L3 verdict: allowed

Resolving egress port policy for [container:id.curl]
* Rule 0 {"matchLabels":{"any:id.curl":""}}: match
  Allows Egress port [{80 tcp}]
1 rules matched
L4 egress verdict: allowed

Resolving ingress port policy for [container:id.httpd]
* Rule 2 {"matchLabels":{"any:id.httpd":""}}: match
  Allows Ingress port [{80 tcp}]
1 rules matched
L4 ingress verdict: allowed

Verdict: allowed
```

**Comment by joestringer** [comment]
Text such as `Rule 2` provides some suggestion about identification or ordering due to the number, but really that number is just an iteration number that's entirely incidental. I think the number should be removed.

**Comment by ianvernon** [comment]
Merged #1810
